test/Performance/DynamoPerformanceTests/PerformanceTestFramework.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Running;
+using BenchmarkDotNet.Validators;
+using Dynamo;
+
+namespace DynamoPerformanceTests
+{
+    public class AllowNonOptimized : ManualConfig
+    {
+        public AllowNonOptimized()
+        {
+            Add(JitOptimizationsValidator.DontFailOnError);
+
+            Add(DefaultConfig.Instance.GetLoggers().ToArray()); // manual config has no loggers by default
+            Add(DefaultConfig.Instance.GetExporters().ToArray()); // manual config has no exporters by default
+            Add(DefaultConfig.Instance.GetColumnProviders().ToArray());
+        }
+    }
+
+    //[Config(typeof(AllowNonOptimized))]
+    public class PerformanceTestFramework : DynamoModelTestBase
+    {
+        
+        protected override void GetLibrariesToPreload(List<string> libraries)
+        {
+            base.GetLibrariesToPreload(libraries);
+
+            libraries.Add("VMDataBridge.dll");
```

Comment:
to replicate actual performance shouldn't we load all libraries the path resolver loads?

Reply:
My initial thought was to create separate test suites, one for tests only using built-in functionality, one for geometry tests, one for excel etc. to get more granularity - so only load the very minimum of libraries to load that particular test suite. This is something we can discuss further.